### DISCUSSION TITLE
Cant get console logs from api routes in NextJS

### DISCUSSION THREAD

**Discussion by amosmachora** [vercel/next.js#59259] [App Router]
I am trying the old trick of `console.log("here")` in my API routes but i can't seem to see the `here` on my logs tab in Vercels dashboard. I found another discussion #21000 that was answered that he logs will show up in like 15 mins but something still doesn't seem right about the answer. Can someone expound on whats that and how i can make them show because im going nuts.

`import { getAccessToken } from "@/util/utils";
import axios from "axios";
import { NextRequest, NextResponse } from "next/server";

const mpesaURL = process.env.MPESA_API_URL;

type RegisterURLParam = {
  ShortCode: "601426" | string;
  ResponseType: "Cancelled" | "Completed";
  ConfirmationURL: string;
  ValidationURL: string;
};

type RegisterURLResponse = {
  OriginatorCoversationID: string;
  ResponseCode: string;
  ResponseDescription: string;
};

export const GET = async (req: NextRequest, res: NextResponse) => {
  console.log("url registration requested");
  const { access_token } = await getAccessToken();

  const registerURLParam = {
    ConfirmationURL: "https://out-of-the-street.vercel.app/api/confirmation",
    ResponseType: "Completed",
    ShortCode: "600999",
    ValidationURL: "https://out-of-the-street.vercel.app/api/validation",
  };

  const response = await axios.post(
    `${mpesaURL}/mpesa/c2b/v1/registerurl`,
    registerURLParam,
    {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${access_token}`,
      },
    }
  );

  return NextResponse.json(response.data);
};
`

**Comment by luannguyenQV** [comment]
I meet the same problems. Did you found a solution for it @amosmachora ?

**Reply by amosmachora** [reply]
actually no. Apparently my API kind of has issues so I'm yet to find out if its nextjs or its my API.

**Reply by mok419** [reply]
Im having this issue too! It is occurring for me in a dynamic api route using [customParameter]

Update: 
I was confused by the new structure of app router and wasn't hitting my API as a result.
Ensure your API is a route.ts file under App/API/[custom] folder with the new format of route handler:
```
export async function GET(
  req: NextApiRequest,
  res: NextApiResponse<ResponseData>
) {
  console.log('hello')
  return new NextResponse(JSON.stringify({ answer: "John Doe" }), {
    status: 200,
  });
}
```

**Comment by JonasDoe** [comment]
I noticed that Next.js creates a static page of those handlers sometimes and - for some obscure reason - drops the `console.log` invocations in doing so. So if this issue only occurs with the production built, you can give it a try with `export const dynamic = 'force-dynamic'` at the top.
